src/statmodels.jl

```diff
@@ -477,6 +477,27 @@ struct TestStat <: Real
 end
 
 show(io::IO, x::TestStat) = @printf(io, "%.2f", x.v)
+TestStat(x::TestStat) = x
+
+float(x::Union{TestStat, PValue}) = float(x.v)
+
+for op in [:(==), :<, :≤, :>, :≥, :(isless), :(isequal)] # isless and < to place nice with NaN
+    @eval begin
+        Base.$op(x::Union{TestStat, PValue}, y::Real) = $op(x.v, y)
+        Base.$op(y::Real, x::Union{TestStat, PValue}) = $op(y, x.v)
+        Base.$op(x1::Union{TestStat, PValue}, x2::Union{TestStat, PValue}) = $op(x1.v, x2.v)
+    end
+end
+
+# necessary to avoid a method ambiguity with isless(::TestStat, NaN)
```

Comment:
So you only need this direction, not with `NaN` as the first argument?

Reply:
Woops, fix incoming